worker/runtime/properties.go

```diff
@@ -2,10 +2,106 @@ package runtime
 
 import (
 	"fmt"
+	"strconv"
+	"strings"
 
 	"code.cloudfoundry.org/garden"
 )
 
+// propertiesToLabels converts a set of properties to a set of labels,
+// splitting up the property values into multiple labels if they would exceed
+// containerd's restriction on the length of the key+value for a single label.
+//
+// each label key is of the form: `key.SEQUENCE_NUMBER`, where SEQUENCE_NUMBER
+// starts from 0 and counts up. for instance, a property `key: ...` may be
+// stored in multiple labels like so:
+//
+// key.0: first chunk of value...
+// key.1: ...second chunk of value...
+// ...
+// key.n: ...last chunk of value
+//
+func propertiesToLabels(properties garden.Properties) (map[string]string, error) {
+	// Hard restriction on the total length of key + value imposed by
+	// containerd on a per-label basis.
+	const maxLabelLen = 4096
+
+	labelSet := map[string]string{}
+	for key, value := range properties {
+		sequenceNum := 0
+		if len(key) > maxLabelLen/2 {
```

Comment:
is the key size limited to 2048 by containerd?

Reply:
No, there's no explicit restriction on key size - all containerd cares about is the combined key + value size.

However, certainly the key + the sequence number suffix (`.0`, `.1`, `.12345`, etc.) can't be longer than 4096, since then no value could be stored. Having a very long (but still <4096) key also means the value needs to be split over many labels, which could cause other problems. For instance, if the key + sequence number is 4095 bytes, and you try to store a 100 byte value, you'd end up needing 100 labels (consuming `400KB` of storage) - given that containerd's label system isn't designed for this, we didn't want to ever let it split this aggressively, so we picked an arbitrary key length limit of half the total length. Given that all of our property names are fixed strings (of ~30 bytes), we felt this was okay.

However, I think this probably deserves a comment, since it is just an arbitrary limit for our property system. Thanks for calling it out!